src/libraries/System.Net.WebClient/src/System/Net/WebClient.cs

```diff
@@ -38,7 +38,8 @@ public class WebClient : Component
         private ProgressData? _progress;
         private IWebProxy? _proxy;
         private bool _proxySet;
-        private int _callNesting; // > 0 if we're in a Read/Write call
+        // It's volatile to avoid optimizing IsBusy away.
+        private volatile int _callNesting; // > 0 if we're in a Read/Write call
```

Comment:
Suggestion: we don't actually need to make it `volatile`, using `Volatile.Read` [here](https://github.com/dotnet/runtime/pull/42076/files#diff-0efdc52d8bf280d45a8f21636c12696dR217) should be enough.

Reply:
Done.